trailblazer/services/analysis_service/utils.py

```diff
@@ -0,0 +1,41 @@
+from trailblazer.constants import TrailblazerStatus
+from trailblazer.dto.analyses_response import UpdateAnalysesResponse
+from trailblazer.dto.analysis_response import AnalysisResponse
+from trailblazer.dto.summaries_response import Summary
+from trailblazer.store.models import Analysis
+
+
+def get_status_count(analyses: list[Analysis], status: TrailblazerStatus) -> int:
+    return len([a for a in analyses if a.status == status])
+
+
+def create_summary(analyses: list[Analysis]) -> Summary:
+    total: int = len(analyses)
+    delivered: int = get_status_count(analyses=analyses, status=TrailblazerStatus.COMPLETED)
+    running: int = get_status_count(analyses=analyses, status=TrailblazerStatus.RUNNING)
+    cancelled: int = get_status_count(analyses=analyses, status=TrailblazerStatus.CANCELLED)
+    failed = get_status_count(analyses=analyses, status=TrailblazerStatus.FAILED)
+    return Summary(
+        order_id=analyses[0].order_id,
```

Comment:
Very minor, but you could consider passing the order_id as only input to this method and fetch analyses inside it. Then you would avoid order_id->analyses->order_id. Feel free to disregard though.

Reply:
Passed the order id, will keep the fetching of the orders outside though